pkg/autodiscovery/providers/env.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package providers
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+const name = "dd_logs_config_custom_configs"
+
+// EnvProvider implements implements the ConfigProvider interface
+// It should be called once at the start of the agent.
+type EnvProvider struct{}
+
+// NewEnvProvider create a EnvProvider searching for
+// configurations in env variable DD_LOGS_CONFIG_CUSTOM_CONFIGS
+func NewEnvProvider() *EnvProvider {
+	return &EnvProvider{}
+}
+
+// Collect get the value of env variable DD_LOGS_CONFIG_CUSTOM_CONFIGS
+// and generate an integrationConfig out of it.
+func (e *EnvProvider) Collect() ([]integration.Config, error) {
```

Comment:
Can we isolate the logic of logs to a dedicated function please and call it from here ?
We could do something like:
```
var collectors := []func()([]integration.Config, error){
  collectLogsConfigs,
  collectMetricsConfigs.
}
func (e *EnvProvider) Collect() ([]integration.Config, error) {  
  for _, collector := range collectors {
    ...
  }
  ...
}

func (e *EnvProvider) collectLogsConfigs() ([]integration.Config, error)
func (e *EnvProvider) collectMetricsConfigs() ([]integration.Config, error)

```